xls/modules/zstd/README.md

```diff
@@ -2,105 +2,192 @@
+| Output Buffer | 0x18 | Keeps the base address for the output buffer, ZSTD Decoder will write the decoded frame into memory starting from this address. |
+
+### Status codes
+
+The following is a list of all available status codes that can be written in the `Status` register.
+
+| Name | Value | Description |
+| ---- | ------- | ----------- |
+| IDLE | 0 | Previous decoding finished successfully. The decoder waits for the configuration and writes to the `Start` register. |
+| RUNNING | 1 | Decoding process is started |
+| READ_CONFIG_OK |2 | Successfully read configuration from the CSRs |
+| FRAME_HEADER_OK | 3 | Successfully decoded frame header |
+| FRAME_HEADER_CORRUPTED | 4 | Frame header data is not valid |
+| FRAME_HEADER_UNSUPPORTED_WINDOW_SIZE | 5 | The `WindowSize` parameter read from the frame header is not supported in the decoder |
+| BLOCK_HEADER_OK | 6 | Successfully read the header of the Zstd data block |
+| BLOCK_HEADER_CORRUPTED | 7 | Block type is `Reserved` |
+| BLOCK_HEADER_MEMORY_ACCESS_ERROR | 8 | Failure in communication with the memory |
+| RAW_BLOCK_OK | 9 | Successfully decoded raw data block |
+| RAW_BLOCK_ERROR | 10 | Failure in communication with the memory |
+| RLE_BLOCK_OK | 11 | Successfully decoded RLE data block |
+
+## Controlling the decoder from the software
+
+The configuration done by the software must be carried out when the decoder is in the `IDLE` state.
+It is the only time when the decoder will be able to take the configuration values from the CSRs and use those in the decoding process.
+
+The software should first read the `Status` register to confirm that the decoder is in the `IDLE` state.
+
+Then, the software has to reserve the memory for the input buffer and write the frame to decode there.
+The address of the buffer should be written into `Input Buffer` register so that the decoder will know where to look for the frame to decode.
+
+The next step is to reserve the memory space for the decoded frame where the Decoder will write the decompressed data.
+The address to that buffer should be written to the `Output Buffer` register.
+
+Finally, it is possible to start the decoding process by writing `1` to the `Start` register.
+This orders the Decoder to read the configuration CSRs and start reading and decoding data stored in the input buffer.
+The Decoder transitions to the `RUNNING` state and then to other states that describe the status of the last operation finished in the decoder (see #status-codes for other possible status codes) which will be visible in the `Status` register.
+
+When the decoding process is finished the Decoder transitions back to the `IDLE` state and signals this on the `Notify` IRQ line.
+The decoded data is stored under the address configured previously in the `Output Buffer` register.
+
+In case an error occurs during the decoding process it is also signaled on the `Notify` IRQ line and the error code is written to the `Status` CSR.
+
 ## ZSTD decoder architecture
```

Comment:
can you arrange the headings so that there is only one H1 in the resulting HTML?

Reply:
can you double-check this?